.github/actions/publish-npm-cli/action.yml

```diff
@@ -0,0 +1,242 @@
+name: 'CLI Publish NPM Package'
+description: 'Package the dotCMS CLI as an NPM project.'
+inputs:
+  ref:
+    description: 'Branch to build from'
+    required: false
+    default: 'master'
+  github-token:
+    description: 'GitHub Token'
+    required: true
+  npm-token:
+    description: 'NPM Token'
+    required: true
+  npm-package-name:
+    description: 'NPM package name'
+    required: false
+    default: 'dotcli'
+  npm-package-scope:
+    description: 'NPM package scope'
+    required: false
+    default: '@dotcms'
+  node-version:
+    description: 'Node.js version'
+    required: false
+    default: '19'
+  workflow-to-search:
+    description: 'Workflow to search for artifacts'
+    required: false
+    default: 'build-test-master.yml'
+  artifact-id:
+    description: 'Artifact id'
+    required: false
+    default: 'cli-artifacts-*'
+
+runs:
+  using: "composite"
+  steps:
+    - name: 'Checkout'
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.ref }}
+
+    - name: 'Set up Node.js'
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node-version }}
+
+    - name: 'Install Jinja2'
+      run: pip install jinja2-cli
+      shell: bash
+
+    - name: 'Download Build Artifact'
+      id: data-download
+      uses: dawidd6/action-download-artifact@v3.1.4
+      with:
+        github_token: ${{ inputs.github-token }}
+        workflow: ${{ inputs.workflow-to-search }}
+        #commit: ${{ github.sha }}
```

Comment:
Is there a possibility when not using commit id that we just use the latest artifact, whatever that is and could pickup the artifact from a previous commit unexpectedly if there was a problem with the artifact from the current.

In reusable-initialize  I turned off specifying the specific workflow (because you cannot specify multiples) and added the workflow-search: true option which would allow it to find it from any workflow then the commit is specified to make sure we do not find in old commits.

Reply:
Now we have a new step to find a previous build if the current fails for somehow.